packages/neuron-ui/src/components/PageContainer/hooks.ts

```diff
@@ -1,34 +1,65 @@
-import { useCallback, useEffect, useState } from 'react'
+import { TFunction } from 'i18next'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import { importedWalletDialogShown } from 'services/localCache'
-import { isDark, openExternal, updateWallet, setTheme as setThemeAPI } from 'services/remote'
+import { isDark, openExternal, setTheme as setThemeAPI, updateWalletStartBlockNumber } from 'services/remote'
 import { Migrate } from 'services/subjects'
 import { getExplorerUrl, isSuccessResponse } from 'utils'
 
+const waitConfirmTime = 5
+
 export const useSetBlockNumber = ({
   firstAddress,
   walletID,
   isMainnet,
   isLightClient,
   isHomePage,
+  initStartBlockNumber,
+  headerTipNumber,
+  t,
 }: {
   firstAddress: string
   walletID: string
   isMainnet: boolean
   isLightClient: boolean
   isHomePage?: boolean
+  initStartBlockNumber?: number
+  headerTipNumber: number
+  t: TFunction
 }) => {
   const [isSetStartBlockShown, setIsSetStartBlockShown] = useState(false)
   const [startBlockNumber, setStartBlockNumber] = useState('')
   const [blockNumberErr, setBlockNumberErr] = useState('')
-  const onChangeStartBlockNumber = useCallback((e: React.SyntheticEvent<HTMLInputElement>) => {
-    const { value } = e.currentTarget
-    const blockNumber = value.replace(/,/g, '')
-    if (Number.isNaN(+blockNumber) || /[^\d]/.test(blockNumber)) {
-      return
+  const isSetLessThanBefore = useMemo(
+    () => !!(startBlockNumber && initStartBlockNumber && +startBlockNumber < initStartBlockNumber),
+    [initStartBlockNumber, startBlockNumber]
+  )
+  const [countdown, setCountdown] = useState(0)
+  const countDecreaseTimeoutRef = useRef<ReturnType<typeof setTimeout>>()
+  const resetCountDown = useCallback(() => {
+    clearTimeout(countDecreaseTimeoutRef.current)
+    setCountdown(waitConfirmTime)
+    const decrement = () => {
+      countDecreaseTimeoutRef.current = setTimeout(() => {
+        setCountdown(v => (v > 0 ? v - 1 : v))
+        decrement()
+      }, 1_000)
     }
-    setStartBlockNumber(blockNumber)
-    setBlockNumberErr('')
-  }, [])
+    decrement()
+  }, [setCountdown, countDecreaseTimeoutRef])
+  const onChangeStartBlockNumber = useCallback(
+    (e: React.SyntheticEvent<HTMLInputElement>) => {
+      const { value } = e.currentTarget
+      const blockNumber = value.replaceAll(',', '')
+      if (Number.isNaN(+blockNumber)) {
+        setCountdown(0)
+        return
+      }
+      setStartBlockNumber(+blockNumber > headerTipNumber ? headerTipNumber.toString() : blockNumber)
+      setBlockNumberErr(+blockNumber > headerTipNumber ? t('set-start-block-number.reset-to-header-tip-number') : '')
+      resetCountDown()
+    },
+    [resetCountDown, setCountdown, initStartBlockNumber, headerTipNumber, t]
+  )
```

Comment:
Mixing a UI hook with a non-UI hook may not be a good practice. Maybe we should split them into two hooks

```jsx
const { current, resetCountdown } = useCountdown(second)
const { blockNumber, setBlockNumber } = useSetBlockNumber

// reset countdown to ensure the user has enough time to confirm 
//  if the block number is right
useEffect(() => {
  resetCountdown();
}, [blockNumber])

<BlockNumberInput 
  onChange={(e) => {
    if(!isValidBlockNumber(e.target.value)) return;
    setBlockNumber(e.target.value)
  }}
/>
```